include/zephyr/bluetooth/mesh/cfg_cli.h

```diff
@@ -1375,9 +1372,1101 @@ int bt_mesh_cfg_mod_sub_del_all_vnd(uint16_t net_idx, uint16_t addr,
+ *  not wait for response and will return immediately after sending the command.
+ *
+ *  @param net_idx   Network index to encrypt with.
+ *  @param addr      Target node address.
+ *  @param elem_addr Element address the model is in.
+ *  @param mod_id    Model ID.
+ *  @param status    Status response parameter.
+ *  @param apps      App index list response parameter. Will be filled with all
+ *                   the returned application key indexes it can fill.
+ *  @param app_cnt   App index list length. Should be set to the capacity of the
+ *                   @c apps list when calling. Will return the number of
+ *                   returned application key indexes upon success.
+ *
+ *  @return 0 on success, or (negative) error code on failure.
+ */
+int bt_mesh_cfg_cli_mod_app_get(uint16_t net_idx, uint16_t addr, uint16_t elem_addr,
+				uint16_t mod_id, uint8_t *status, uint16_t *apps, size_t *app_cnt);
+
+/** @brief Get a list of all applications bound to a vendor model on the target
+ *         node.
+ *
+ *  This method can be used asynchronously by setting @p status
+ *  and ( @p apps or @p app_cnt ) as NULL. This way the method will
+ *  not wait for response and will return immediately after sending the command.
+ *
+ *  @param net_idx   Network index to encrypt with.
+ *  @param addr      Target node address.
+ *  @param elem_addr Element address the model is in.
+ *  @param mod_id    Model ID.
+ *  @param cid       Company ID of the model.
+ *  @param status    Status response parameter.
+ *  @param apps      App index list response parameter. Will be filled with all
+ *                   the returned application key indexes it can fill.
+ *  @param app_cnt   App index list length. Should be set to the capacity of the
+ *                   @c apps list when calling. Will return the number of
+ *                   returned application key indexes upon success.
+ *
+ *  @return 0 on success, or (negative) error code on failure.
+ */
+int bt_mesh_cfg_cli_mod_app_get_vnd(uint16_t net_idx, uint16_t addr, uint16_t elem_addr,
+				    uint16_t mod_id, uint16_t cid, uint8_t *status, uint16_t *apps,
+				    size_t *app_cnt);
+
+/** @brief Get publish parameters for a SIG model on the target node.
+ *
+ *  This method can be used asynchronously by setting @p status and
+ *  @p pub as NULL. This way the method will not wait for response
+ *  and will return immediately after sending the command.
+ *
+ *  @param net_idx   Network index to encrypt with.
+ *  @param addr      Target node address.
+ *  @param elem_addr Element address the model is in.
+ *  @param mod_id    Model ID.
+ *  @param pub       Publication parameter return buffer.
+ *  @param status    Status response parameter.
+ *
+ *  @return 0 on success, or (negative) error code on failure.
+ */
+int bt_mesh_cfg_cli_mod_pub_get(uint16_t net_idx, uint16_t addr, uint16_t elem_addr,
+				uint16_t mod_id, struct bt_mesh_cfg_mod_pub *pub, uint8_t *status);
```

Comment:
Should we also deprecate "struct bt_mesh_cfg_mod_pub" and rename it to "struct bt_mesh_cfg_cli_mod_pub"? This will be consistent with the functions.